rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -71,22 +71,28 @@ where
     /// # Errors
     /// [`struct@Error`] of kind [`InvalidClientId`](ErrorKind::InvalidClientId)
     /// if the Client Id of the [`ManagedClient`] isn't valid as a topic token.
+    /// # Panics
+    /// Panics if the options for the underlying command invokers or receivers cannot be built. Not possible since
+    /// the options are statically generated.
     #[allow(clippy::needless_pass_by_value)]
     pub fn new(
         application_context: ApplicationContext,
         client: C,
         options: ClientOptions,
     ) -> Result<Self, Error> {
+        // THIS IS A TEMPORARY FIX. WORKAROUND FOR THE FACT THAT CODEGEN PANICS ON INVALID CLIENT ID
+        // INSTEAD OF RETURNING AN ERROR
         if !Self::is_valid_replacement(client.client_id()) {
             return Err(ErrorKind::InvalidClientId(client.client_id().to_string()).into());
         }
+
         let command_options = CommandInvokerOptionsBuilder::default()
             .topic_token_map(HashMap::from([(
                 "connectorClientId".to_string(),
                 client.client_id().to_string(),
             )]))
             .build()
-            .map_err(ErrorKind::from)?;
+            .expect("Builder cannot fail as there is no validation function");
```

Comment:
the builder can (technically) fail, and there is a validation function at some layer. However, it can't fail _here_ because everything is statically generated (from the DTDL) or validated before this point (with the client id check we have)
```suggestion
            .expect("Builder cannot fail as input is statically generated or validated before this point");
```

Reply:
sync'd offline, this specific builder doesn't actually validate anything, so the original message is correct